app/utils/file/file_picker/index.ts

```diff
@@ -0,0 +1,340 @@
+                            id: 'mobile.permission_denied_dismiss',
+                            defaultMessage: "Don't Allow",
+                        }),
+                    },
+                    {
+                        text: this.intl.formatMessage({
+                            id: 'mobile.permission_denied_retry',
+                            defaultMessage: 'Settings',
+                        }),
+                        onPress: () => AndroidOpenSettings.appDetailsSettings(),
+                    },
+                ]);
+                return false;
+            }
+            default: return true;
+        }
+
+        return false;
+    };
+
+    private buildUri = async (doc: DocumentPickerResponse) => {
+        let uri: string = doc.uri;
+
+        if (Platform.OS === 'android') {
+            // For android we need to retrieve the realPath in case the file being imported is from the cloud
+            const newUri = await ShareExtension.getFilePath(doc.uri);
+            uri = newUri?.filePath;
+            if (uri === undefined) {
+                return {doc: undefined};
+            }
+        }
+
+        // Decode file uri to get the actual path
+        doc.uri = decodeURIComponent(uri);
+        return {doc};
+    };
+
+    attachFileFromCamera = async (customOptions?: CameraOptions) => {
+        let options = customOptions;
+        if (!options) {
+            options = {
+                quality: 0.8,
+                videoQuality: 'high',
+                mediaType: 'photo',
+                saveToPhotos: true,
+            };
+        }
+
+        const hasCameraPermission = await this.hasPhotoPermission('camera');
+        if (hasCameraPermission) {
+            launchCamera(options, async (response: ImagePickerResponse) => {
+                StatusBar.setHidden(false);
+
+                if (response.errorCode || response.didCancel) {
+                    return;
+                }
+
+                const files = await this.getFilesFromResponse(response);
+                await this.prepareFileUpload(files);
+            });
+        }
+    };
+
+    attachFileFromFiles = async (browseFileType?: string) => {
+        const hasPermission = await this.hasStoragePermission();
+        const fileType = browseFileType ?? Platform.select({ios: 'public.item', android: '*/*'})!;
+
+        if (hasPermission) {
+            try {
+                const docResponse = (await DocumentPicker.pickMultiple({type: [fileType]}));
+                const proDocs = docResponse.map(async (d: DocumentPickerResponse) => {
+                    const {doc} = await this.buildUri(d);
+                    return doc;
+                });
+
+                const docs = await Promise.all(proDocs) as unknown as DocumentPickerResponse[];
+
+                await this.prepareFileUpload(docs);
+            } catch (error) {
+                // Do nothing
+            }
+        }
+    };
+
+    attachFileFromPhotoGallery = async () => {
```

Comment:
we should be able to define `selectionLimit` see https://github.com/react-native-image-picker/react-native-image-picker#options

to determine how many images can be picked from the photo gallery

Reply:
done